pkg/cephmgr/osd/agent.go

```diff
@@ -854,6 +852,38 @@ func associateOSDIDsWithMetadataDevice(etcdClient etcd.KeysAPI, nodeID, name, id
 	return err
 }
 
+func markOSDAsApplied(context *clusterd.Context, config *osdConfig) error {
+	if context.EtcdClient != nil {
```

Comment:
nit: if you return when context.EtcdCliet == nil, you could save some indentation for the rest of the block